### ISSUE TITLE
gcp-auth not working in a GCE instance with a service account

### ISSUE THREAD

**Issue by amontalban** [kubernetes/minikube#12349]
**Steps to reproduce the issue:** 

1. Create a GCE instance with a service account that has permissions to pull images from a private registry.
2. Pods will be in `ImagePullBackOff`.
3. Using `gcloud auth configure-docker` and then doing `docker pull gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}` works from the GCE instance.

Output:

```
ubuntu@lb-minikube-dev-test:~$ minikube version
minikube version: v1.22.0
commit: a03fbcf166e6f74ef224d4a63be4277d017bb62e
ubuntu@lb-minikube-dev-test:~$ minikube start
* minikube v1.22.0 on Ubuntu 20.04 (amd64)
* Using the docker driver based on user configuration
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Creating docker container (CPUs=12, Memory=50176MB) ...
* Preparing Kubernetes v1.21.2 on Docker 20.10.7 ...
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Verifying Kubernetes components...
  - Using image gcr.io/k8s-minikube/storage-provisioner:v5
* Enabled addons: storage-provisioner, default-storageclass
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
ubuntu@lb-minikube-dev-test:~$ minikube addons enable gcp-auth

* Exiting due to MK_CREDENTIALS_NOT_NEEDED: It seems that you are running in GCE, which means authentication should work without the GCP Auth addon. If you would still like to authenticate using a credentials file, use the --force flag.
```

`kubect describe pod`:

```
Events:
  Type     Reason     Age   From               Message
  ----     ------     ----  ----               -------
  Normal   Scheduled  55s   default-scheduler  Successfully assigned default/service-59675cb7db-c2vsm to minikube
  Normal   Pulling    52s   kubelet            Pulling image "gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}"
  Warning  Failed     18s   kubelet            Failed to pull image "gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}": rpc error: code = Unknown desc = Error response from daemon: unauthorized: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication
  Warning  Failed     18s   kubelet            Error: ErrImagePull
  Normal   BackOff    17s   kubelet            Back-off pulling image "gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}"
  Warning  Failed     17s   kubelet            Error: ImagePullBackOff
```

`docker pull`:

```
ubuntu@lb-minikube-dev-test:~/iac-kubernetes$ gcloud auth configure-docker

WARNING: Your config file at [/home/ubuntu/.docker/config.json] contains these credential helper entries:

{
  "credHelpers": {
    "gcr.io": "gcloud",
    "us.gcr.io": "gcloud",
    "eu.gcr.io": "gcloud",
    "asia.gcr.io": "gcloud",
    "staging-k8s.gcr.io": "gcloud",
    "marketplace.gcr.io": "gcloud"
  }
}
Adding credentials for all GCR repositories.
WARNING: A long list of credential helpers may cause delays running 'docker build'. We recommend passing the registry name to configure only the registry you are using.
gcloud credential helpers already registered correctly.
ubuntu@lb-minikube-dev-test:~/iac-kubernetes$ 
ubuntu@lb-minikube-dev-test:~/iac-kubernetes$ docker pull gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}
develop: Pulling from {PROJECT_ID}/{IMAGE}
69692152171a: Pull complete 
ce2b89b60818: Pull complete 
6584437267ff: Pull complete 
b6500b56ee97: Pull complete 
c54f457aa338: Pull complete 
8062c80dc5ed: Pull complete 
6840cf5d5ea7: Pull complete 
1749ee69000a: Pull complete 
b690f06f38d6: Pull complete 
adda2d37bfd4: Pull complete 
cc7a536eef97: Pull complete 
c957e3cede4a: Pull complete 
6b03e330803d: Pull complete 
Digest: sha256:626c3f1064175b281599fe7ae044c51a6c4381c4f58d16e39d3f63091ea61332
Status: Downloaded newer image for gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}
gcr.io/{PROJECT_ID}/{IMAGE}:{TAG}
```

**Comment by sharifelgamal** [comment]
The source of this issue is that when running `minikube addons enable gcp-auth` in GCE, we were exiting out before adding in the pull secret from `configure-docker`. This was fixed in #12124 and will be included in the upcoming release this week.

The workaround in the meantime to use the `--force` flag with the addons command, which can have knock-on effects but will fix the issue you're currently seeing.

**Comment by amontalban** [comment]
Thank you very much @sharifelgamal, looking forward for the upcoming release!
